bookwyrm/views/author.py

```diff
@@ -12,20 +12,23 @@
 from bookwyrm.activitypub import ActivitypubResponse
 from bookwyrm.connectors import connector_manager
 from bookwyrm.settings import PAGE_LENGTH
-from bookwyrm.views.helpers import is_api_request
+from bookwyrm.views.helpers import is_api_request, maybe_redirect_local_path
 
 
 # pylint: disable= no-self-use
 class Author(View):
     """this person wrote a book"""
 
-    def get(self, request, author_id):
+    def get(self, request, author_id, slug=None):
```

Comment:
So this is failing the lint because of the unused arg. Is this a case where we just do a linter ignore, or is there some other way we can use `{% url %}` to construct a URL and then discard a param?

Reply:
If I understand correctly, it needs to be there because the slug is a url parameter and so the view has to receive the parameter for the url to work. Can the slug be not a parameter? Like if the regex was just `r"/s/[-_a-z0-9]+"` instead of `r"/s/(?P<slug>[-_a-z0-9]+)"`. If that works, then the parameter wouldn't be necessary in the view.

But if that doesn't work I'd definitely just do a pylint disable on it.

Reply:
now that I think about it, that idea probably won't work because, as you pointed out right there, it's needed for the `{% url %}` tag

Reply:
Do you think it's reasonable to append the slug manually everywhere we call `{% url %}`, like this? 
```
href="{% url 'book' book.id %}/{{ book.name | slugify}}"
```
Does that feel too hacky?

Reply:
Hmm, I think that seems fine to me? I feel like either solution is kind of odd in its own way, and this one at least keeps the oddness in slightly fewer places?

Reply:
I think I'll go with the linting annotations because I was running into complexities in spots where we have `{% url 'book' book.id slug as book_url %}` and it feels like it would be messy to have to compose the URL manually.

Reply:
Sounds good! Thanks for thinking it through with me